mtags/shared/src/main/scala-3/scala/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -0,0 +1,538 @@
+    options: List[String] = Nil,
+    search: SymbolSearch = EmptySymbolSearch,
+    ec: ExecutionContextExecutor = ExecutionContext.global,
+    sh: Option[ScheduledExecutorService] = None,
+    config: PresentationCompilerConfig = PresentationCompilerConfigImpl()
+) extends PresentationCompiler {
+
+  def this() = this(Nil, Nil)
+
+  import InteractiveDriver._
+
+  val compilerAccess: CompilerAccess[StoreReporter, InteractiveDriver] = {
+    Scala3CompilerAccess(
+      config,
+      sh,
+      () => new Scala3CompilerWrapper(newDriver)
+    )(
+      ec
+    )
+  }
+
+  def newDriver: InteractiveDriver = {
+    val defaultFlags = List("-color:never")
+    val settings =
+      options ::: defaultFlags ::: "-classpath" :: classpath.mkString(
+        File.pathSeparator
+      ) :: Nil
+    new InteractiveDriver(settings)
+  }
+
+  def complete(params: OffsetParams): CompletableFuture[CompletionList] = {
+    compilerAccess.withInterruptableCompiler(emptyCompletion(), params.token) {
+      access =>
+        val driver = access.compiler()
+        val uri = params.uri
+        driver.run(uri, params.text)
+        implicit def ctx: Context = driver.currentCtx
+
+        val pos = sourcePosition(driver, params, uri)
+        val items = driver.compilationUnits.get(uri) match {
+          case Some(unit) =>
+            Completion.completions(pos)(ctx.fresh.setCompilationUnit(unit))._2
+          case None => Nil
+        }
+
+        new CompletionList(
+          /*isIncomplete = */ false,
+          items.map(completionItem).asJava
+        )
+    }
+  }
+
+  def definition(params: OffsetParams): CompletableFuture[DefinitionResult] = {
+    compilerAccess.withNonInterruptableCompiler(
+      DefinitionResultImpl.empty,
+      params.token
+    ) { access =>
+      val driver = access.compiler()
+      implicit def ctx: Context = driver.currentCtx
+      val uri = params.uri
+      driver.run(uri, params.text)
+      val pos = sourcePosition(driver, params, uri)
+      val path = Interactive.pathTo(driver.openedTrees(uri), pos)
+      val definitions = Interactive.findDefinitions(path, pos, driver).toList
+
+      DefinitionResultImpl(
+        "",
+        definitions.flatMap(d => location(d.namePos)).asJava
+      )
+
+    }
+  }
+
+  def shutdown(): Unit = {
+    compilerAccess.shutdown()
+  }
+
+  def restart(): Unit = {
+    compilerAccess.shutdownCurrentCompiler()
+  }
+
+  def diagnosticsForDebuggingPurposes(): ju.List[String] = List[String]().asJava
+
+  // TODO NOT IMPLEMENTED
+  def semanticdbTextDocument(
+      filename: String,
+      code: String
+  ): CompletableFuture[Array[Byte]] = {
+    CompletableFuture.completedFuture(
+      List[Byte]().toArray
+    )
+  }
+
+  // TODO NOT IMPLEMENTED
+  def completionItemResolve(
+      item: CompletionItem,
+      symbol: String
+  ): CompletableFuture[CompletionItem] = {
+    CompletableFuture.completedFuture(
+      null
+    )
+  }
+
+  // TODO NOT IMPLEMENTED
```

Comment:
Some of the features are not yet implemented, but Metals work perfectly well without them, so I would rather merge this without those features and work on them later. Plus it can be later worked on by multiple people when needed.